apps/web/src/views/Swap/V3Swap/hooks/useSendSwapTransaction.ts

```diff
@@ -214,15 +221,20 @@ export default function useSendSwapTransaction(
             } else {
               // otherwise, the error was unexpected and we need to convey that
               logger.warn(
-                'Swap failed',
+                error instanceof EstimateGasExecutionError ? 'Estimate failed' : 'Swap failed',
```

Comment:
Estimate should be done here?
https://github.com/pancakeswap/pancake-frontend/pull/9399/files#diff-83218ea7d05848674a6cd69042376956fd3dfcb9f694f65f7954867778363739R100

Reply:
And let's add an identifier to link the estimation and execution

Reply:
so we can locate both gas estimation and swap execution for a single tx. Let's log the estimated gas cost and the actual gas specified for the execution.

Reply:
> Estimate should be done here? [#9399 (files)](https://github.com/pancakeswap/pancake-frontend/pull/9399/files#diff-83218ea7d05848674a6cd69042376956fd3dfcb9f694f65f7954867778363739R100)

you are right, firstly I think `sendTransactionAsync` will do `prepareTransactionRequest` internal, which will estimate again, but I forgot we have passed `gas` parameter.

Reply:
So, I think we should no need to link estimate and execution logs? as only will be one of them failed.

or you want to log normal actions?

Reply:
> so we can locate both gas estimation and swap execution for a single tx. Let's log the estimated gas cost and the actual gas specified for the execution.

for `TransactionExecutionError`, the error log already include `gas` info

Reply:
the error log only shows how much gas actually used right? I'm suggesting linking the two logs is because we can compare the how much gas is estimated and how much gas we actually used. To see if we need to increase the margin for some cases

Reply:
https://github.com/wevm/viem/blob/main/src/actions/public/multicall.ts#L190-L196

the gas in error log is request params

Reply:
ah ok. Then should be good